src/plugins/intel_gpu/src/kernel_selector/kernels/convolution/convolution_kernel_b_fs_yx_fsv16_1x1.cpp

```diff
@@ -215,40 +241,80 @@ JitConstants ConvolutionKernel_b_fs_yx_fsv16_1x1::GetJitConstants(const convolut
-        for (auto& t : fused_op.tensors) {
-            if (t.PitchesDifferFromLogicalDims()) {
-                padded_output = true;
+    if (!params.has_dynamic_inputs()) {
+        jit.AddConstant(MakeJitConstant("PADDED_INPUT", params.inputs[0].X().pad.Total() != 0));
+
+        bool padded_output = params.outputs[0].X().pad.Total() != 0;
+        bool non_unit_fused_op_spatial = false;
+
+        // Set padded_output to true when fused inputs have paddings to have correct blocked loads
+        for (auto& fused_op : params.fused_ops) {
+            for (auto& t : fused_op.tensors) {
+                if (t.PitchesDifferFromLogicalDims()) {
+                    padded_output = true;
+                }
+                if ((t.X().v > 1) ||
+                    (t.Y().v > 1) ||
+                    (t.Z().v > 1) ||
+                    (t.W().v > 1)) {
+                    non_unit_fused_op_spatial = true;
+                }
             }
-            if ((t.X().v > 1) ||
-                (t.Y().v > 1) ||
-                (t.Z().v > 1) ||
-                (t.W().v > 1)) {
-                non_unit_fused_op_spatial = true;
+        }
+
+        jit.AddConstant(MakeJitConstant("PADDED_OUTPUT", padded_output));
+        jit.AddConstant(MakeJitConstant("NON_UNIT_FUSED_OP_SPATIAL", non_unit_fused_op_spatial));
+
+        jit.AddConstant(MakeJitConstant("IC_BLOCKS", CeilDiv(params.inputs[0].Feature().v, tuning_data.feature_block_size)));
+        if (params.outputs[0].Feature().v % tuning_data.feature_block_size != 0) {
+            jit.AddConstant(MakeJitConstant("OUTPUT_LEFTOVERS", 1));
+        }
+        if (params.inputs[0].Feature().v % tuning_data.feature_block_size != 0) {
+            jit.AddConstant(MakeJitConstant("INPUT_LEFTOVERS", 1));
+        }
+    } else {
+        DimensionAccessHelperJit input0_dims(params.inputs[0]);
+        DimensionAccessHelperJit input0_padded_dims(params.inputs[0], true);
+        DimensionAccessHelperJit output_dims(params.outputs[0]);
+        DimensionAccessHelperJit output_padded_dims(params.outputs[0], true);
+
+        const auto padded_input = "(" + input0_padded_dims.x_pad().first + "+" + input0_padded_dims.x_pad().first + ") != 0";
+        jit.AddConstant(MakeJitConstant("PADDED_INPUT", padded_input));
+
+        const auto padded_output = "(" + output_padded_dims.x_pad().first + "+" + output_padded_dims.x_pad().first + ") != 0";
+        jit.AddConstant(MakeJitConstant("PADDED_OUTPUT", padded_output));
+
+        // In shape agnostic kernel, the fused shape cannot be specified at build time or run time.
+        // Currently simply check whether fused_op is dynmaic. Need to further follow up like static behavior.
+        bool non_unit_fused_op_spatial = false;
+        for (auto& fused_op : params.fused_ops) {
+            for (auto& t : fused_op.tensors) {
+                if (t.is_dynamic()) {
+                    non_unit_fused_op_spatial = true;
+                    break;
+                }
```

Comment:
```suggestion
                if (t.is_dynamic()) {
                    non_unit_fused_op_spatial = true;
                    break;
                } else {
                     if ((t.X().v > 1) ||
                        (t.Y().v > 1) ||
                        (t.Z().v > 1) ||
                        (t.W().v > 1)) {
                        non_unit_fused_op_spatial = true;
                }
```
Could you update it to check tensor size when tensor is static?

Reply:
Applied